src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs

```diff
@@ -1823,6 +1823,20 @@ public void TestEnumDeclarationFromSymbol()
 }");
         }
 
+        [Fact]
+        public void TestEnumWithUnderlyingTypeFromSymbol()
+        {
+            VerifySyntax<EnumDeclarationSyntax>(
+                    Generator.Declaration(
+                        _emptyCompilation.GetTypeByMetadataName("System.Security.SecurityRuleSet")),
+@"public enum SecurityRuleSet : global::System.Byte
```

Comment:
nope: this won't parse.  the base type must be an actual primitive like `byte`.  we'll likely have to special case this in the generator. :)

Reply:
Ok, I'll look into it. :)

Reply:
Interesting, when I tried it in sharplab it did parse it correctly, but makes sense to return it like an actual primitive.
https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEDMACOA7ArmDAMQHtiMQMBzAG2OAmpBBQEYA6AIQE94MBvDAF8gA=

Reply:
Fixed.